ipywidgets/static/widgets/js/manager.js

```diff
@@ -280,11 +280,26 @@ define([
         /**
          * Handle when a comm is opened.
          */
-        return this.create_model({
-            model_name: msg.content.data.model_name,
-            model_module: msg.content.data.model_module,
-            comm: comm,
-        }).catch(utils.reject("Couldn't create a model.", true));
+        // For < 4.x
+        var model;
+        if (msg.content.data.model_name) {
```

Comment:
The reasoning is the following: there are two types of nbextensions, the ones that are independent of the kernel, like some nbconvert button in the toolbar, and the ones that have a piece of code on the kernel side.
The latter is more complicated because you may have multiple kernel registered having different versions of the same nbextension like ipywidgets.

I think that for each registered kernel, we should have the Javascript in a different directory, (one per installed kernelspec). The requirejs path would be automatically prefixed with a string corresponding to the current kernel.